src/loaders/workspace_downloader/workspace_downloader.py

```diff
@@ -0,0 +1,319 @@
+            break
+        cfn = os.path.join(conf.job_dir, "workdir/tmp", upa)
+        # upa file is downloaded to cfn
+        conf.asu.get_assembly_as_fasta({"ref": upa.replace("_", "/"), "filename": upa})
+
+        # each upa in output_dir as a seperate directory
+        dstd = os.path.join(conf.pth, upa)
+        os.makedirs(dstd, exist_ok=True)
+
+        dst = os.path.join(dstd, f"{upa}.fa")
+        # copy downloaded upa file to output_dir. Hard link might cause invalid cross-device link problem
+        shutil.copy(cfn, dst)
+
+        metafile = os.path.join(dstd, f"{upa}.meta")
+        # save meta file with relevant object_info
+        json.dump(_process_object_info(obj_info), open(metafile, "w"), indent=2)
+
+        print("Completed %s" % (upa))
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="PROTOTYPE - Download genome files from the workspace service (WSS)."
+    )
+
+    required = parser.add_argument_group("required named arguments")
+    optional = parser.add_argument_group("optional arguments")
+
+    # Required flag argument
+    required.add_argument(
+        "--workspace_id",
+        required=True,
+        type=int,
+        help="Workspace addressed by the permanent ID",
+    )
+    required.add_argument(
+        "--project_dir",
+        type=str,
+        help="Path points to Collections repository",
+    )
+
+    # Optional argument
+    optional.add_argument(
+        "--root_dir",
+        type=str,
+        default=loader_common_names.ROOT_DIR,
+        help="Root directory.",
+    )
+    optional.add_argument(
+        "--output_dir",
+        type=str,
+        help="Output directoy to save genome objects.",
+    )
+    optional.add_argument(
+        "--job_dir",
+        type=str,
+        help="Job directoy of source link",
+    )
+    optional.add_argument(
+        "--source", type=str, default="WS", help="Source of data (default: WS)"
+    )
+    optional.add_argument(
+        "--workers",
+        type=int,
+        default=5,
+        help="Number of workers for multiprocessing",
+    )
+    optional.add_argument(
+        "--overwrite", action="store_true", help="Overwrite existing files."
+    )
+    optional.add_argument(
+        "--batch_size", type=int, default=10000, help="Batch size of object id"
+    )
+
+    args = parser.parse_args()
+
+    (
+        workspace_id,
+        project_dir,
+        root_dir,
+        output_dir,
+        job_dir,
+        source,
+        workers,
+        overwrite,
+        batch_size,
+    ) = (
+        args.workspace_id,
+        args.project_dir,
+        args.root_dir,
+        args.output_dir,
+        args.job_dir,
+        args.source,
+        args.workers,
+        args.overwrite,
+        args.batch_size,
+    )
+
+    if source not in SOURCE:
+        raise ValueError(f"Unexpected source. Currently supported sources: {SOURCE}")
+
+    uid = loader_helper.get_id()
+    username = loader_helper.get_username()
```

Comment:
Per the 1pm, `job_dir` is the directory where the SDK callback server writes the various files needed for an SDK run, including the output of the run.

We discussed changing `job_dir` from an input argument to always writing to `<root_dir>/job_dir/<linux_user_name>`. This means that `os.link` should always work as well.

The script should ensure that the `<linux_user_name>` dir is only readable and writable by the current user as tokens will be stored in that directory

Reply:
tokens here you mean UPA right, not "KB_AUTH_TOKEN"

Reply:
No, SDK runs store the user's token in the job dir. This usually happens automatically. I'm not 100% sure the callback server does that here but let's be safe just in case, or just in case the behavior changes.

Reply:
For example
```
~/github/kbaseapps/AssemblyUtil/test_local/workdir$ ls
config.properties  coverage.xml  test.cfg  test_coverage  token
```

Reply:
I see. Now is clear!

Reply:
According to 1pm discussion, job_dir now is predefined at /global/cfs/cdirs/kbase/collections/job_dir

Reply:
Can you `os.chmod` the jobdir to make sure it's -rwx for everyone except the user

Reply:
ah, you mean it's **not** -rwx for everyone except the user, right?

Reply:
I mean not one except the user can read, write, or execute it

Reply:
Updated!

Reply:
Can we change the name of `JOB_DIR` to `sdk_job_dir` to avoid confusion with the task farmer job dirs?

Reply:
Updated!

Reply:
:+1: